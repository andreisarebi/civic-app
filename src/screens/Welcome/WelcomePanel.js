import { StyleSheet, Text, View } from 'react-native';
import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'react-native-elements';
import colors from '../../styles/colors';
import { formTypes } from '../../auth/redux';
import SocialButton from '../../auth/SocialButton';

class WelcomePanel extends React.Component {
  static propTypes = {
    switchFormType: PropTypes.func,
    formType: PropTypes.string,
    navigate: PropTypes.func,
  };

  sharedButtonProps = {
    borderRadius: 3,
    containerViewStyle: styles.buttonContainer,
    fontSize: 18,
  };

  defaultView = (
    <View style={styles.defaultViewContainer}>
      <Button
        {...this.sharedButtonProps}
        color={colors.white}
        backgroundColor={colors.red}
        title="Register"
        onPress={() => this.props.switchFormType(formTypes.SIGN_UP)}
      />
      <Button
        {...this.sharedButtonProps}
        color={colors.black}
        backgroundColor={colors.offWhite}
        title="Sign In"
        onPress={() => this.props.switchFormType(formTypes.LOGIN)}
      />
    </View>
  );

  expandedView = formType => {
    const config = (type => {
      switch (type) {
        case formTypes.INITIAL:
          return {};
        case formTypes.LOGIN:
          return {
            preposition: 'in',
            switchText: 'Don\'t have an account yet? Register',
            otherFormType: formTypes.SIGN_UP,
          };
        case formTypes.SIGN_UP:
          return {
            preposition: 'up',
            switchText: 'Have an account? Sign in',
            otherFormType: formTypes.LOGIN,
          };
        default:
          return {};
      }
    })(formType);

    return (
      <View style={styles.expandedViewContainer}>
        <SocialButton
          type="google"
          title={`Sign ${config.preposition} with Google`}
          style={styles.social}
        />
        <SocialButton type="facebook" title="Continue with Facebook" style={styles.social} />
        <Text style={styles.text}>or</Text>
        <SocialButton
          type="email"
          title={`Sign ${config.preposition} with email`}
          style={styles.social}
          onPress={this.goToCredentials}
        />
        <Text
          style={[styles.text, styles.switchText]}
          onPress={() => this.props.switchFormType(config.otherFormType)}
        >
          {config.switchText}
        </Text>
      </View>
    );
  };

  goToCredentials = () => {
    this.props.navigate('Credentials');
  };

  render() {
    return (
      <View style={styles.container}>
        {this.props.formType === formTypes.INITIAL
          ? this.defaultView
          : this.expandedView(this.props.formType)}
      </View>
    );
  }
}

var styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    position: 'absolute',
    bottom: 0,
    backgroundColor: colors.white,
    borderTopLeftRadius: 25,
    borderTopRightRadius: 25,
  },
  buttonContainer: {
    flex: 1,
    marginTop: 20,
    marginBottom: 20,
    borderRadius: 3,
    shadowColor: colors.black,
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.5,
    shadowRadius: 3,
  },
  defaultViewContainer: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center',
  },
  expandedViewContainer: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    padding: 20,
  },
  social: {
    marginBottom: 8,
    width: '100%',
    maxWidth: 322,
  },
  text: {
    textAlign: 'center',
    fontSize: 18,
    paddingTop: 10,
    paddingBottom: 18,
  },
  switchText: {
    paddingBottom: 0,
  },
});

export default WelcomePanel;
