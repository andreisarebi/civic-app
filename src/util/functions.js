import distanceInWordsStrict from 'date-fns/distance_in_words_strict';

export const distanceInWordsFromNow = (date) => {
  return distanceInWordsStrict(date, Date.now());
}
