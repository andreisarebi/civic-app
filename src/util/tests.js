import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { Provider } from 'react-redux';
import store from '../../App/configureStore';

export const createSnapshot = (Component, props) => {
  const renderer = new ShallowRenderer();
  renderer.render(<Component {...props} />);
  return renderer.getRenderOutput();
};

export const createConnectedSnapshot = (Component, props) => {
  const renderer = new ShallowRenderer();
  renderer.render(<Provider store={store}><Component {...props} /></Provider>);
  return renderer.getRenderOutput();
};
