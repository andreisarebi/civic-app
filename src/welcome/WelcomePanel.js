import { StyleSheet, View } from 'react-native';
import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'react-native-elements';
import colors from '../styles/colors';

var styles = StyleSheet.create({
  container: {
    backgroundColor: colors.white,
    borderTopLeftRadius: 25,
    borderTopRightRadius: 25,
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center',
  },
  buttonContainer: {
    flex: 1,
    marginTop: 20,
    marginBottom: 20,
    borderRadius: 3,
    shadowColor: colors.black,
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.5,
    shadowRadius: 3,
  },
});

class WelcomePanel extends React.Component {
  static propTypes = {
    onButtonPress: PropTypes.func,
  };

  sharedButtonProps = {
    borderRadius: 3,
    containerViewStyle: styles.buttonContainer,
    fontSize: 18,
  };

  render() {
    return (
      <View style={styles.container}>
        <Button
          {...this.sharedButtonProps}
          color={colors.white}
          backgroundColor={colors.red}
          title="Register"
          onPress={this.props.onButtonPress}
        />
        <Button
          {...this.sharedButtonProps}
          color={colors.black}
          backgroundColor={colors.offWhite}
          title="Sign In"
          onPress={this.props.onButtonPress}
        />
      </View>
    );
  }
}

export default WelcomePanel;
