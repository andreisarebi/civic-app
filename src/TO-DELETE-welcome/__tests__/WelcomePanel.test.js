import { createSnapshot } from '../../util/tests';
import WelcomePanel from '../WelcomePanel';

describe('WelcomePanel', () => {
  it('renders correctly', () => {
    const snapshot = createSnapshot(WelcomePanel);
    expect(snapshot).toMatchSnapshot();
  });
});
