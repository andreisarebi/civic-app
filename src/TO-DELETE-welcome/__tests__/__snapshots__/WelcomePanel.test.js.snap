// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomePanel renders correctly 1`] = `
<AnimatedComponent
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "borderTopLeftRadius": 25,
        "borderTopRightRadius": 25,
        "flexDirection": "row",
      },
      Object {
        "bottom": 0,
        "position": "absolute",
      },
    ]
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "padding": 20,
      }
    }
  >
    <SocialButton
      onLongPress={undefined}
      onPress={undefined}
      style={
        Object {
          "marginBottom": 8,
          "maxWidth": 322,
          "width": "100%",
        }
      }
      title="Sign undefined with Google"
      type="google"
    />
    <SocialButton
      onLongPress={undefined}
      onPress={undefined}
      style={
        Object {
          "marginBottom": 8,
          "maxWidth": 322,
          "width": "100%",
        }
      }
      title="Continue with Facebook"
      type="facebook"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 18,
          "paddingBottom": 18,
          "paddingTop": 10,
          "textAlign": "center",
        }
      }
    >
      or
    </Text>
    <SocialButton
      onPress={[Function]}
      style={
        Object {
          "marginBottom": 8,
          "maxWidth": 322,
          "width": "100%",
        }
      }
      title="Sign undefined with email"
      type="email"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Array [
          Object {
            "fontSize": 18,
            "paddingBottom": 18,
            "paddingTop": 10,
            "textAlign": "center",
          },
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    />
  </Component>
</AnimatedComponent>
`;
